### ISSUE TITLE
Possible loading issue under Minitest 5.19.0

### ISSUE THREAD

**Issue by dazuma** [minitest/minitest-rg#17]
Minitest 5.19.0 was released today (July 26, 2023), and it has a behavior change that seems to affect minitest-rg loading. Specifically, when we update to minitest 5.19.0, the minitest-rg plugin (when we `require "minitest/rg"` in our test helper) now raises:

```
[2023-07-26 20:55:44  INFO]  exec: ["bundle", "exec", "rake", "test"]
/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/minitest-rg-5.2.0/lib/minitest/rg.rb:4:in `<top (required)>': uninitialized constant Minitest::RG (NameError)

Minitest::RG.rg!
        ^^^^
	from <internal:/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /home/runner/work/google-cloud-ruby/google-cloud-ruby/google-analytics-admin/test/helper.rb:23:in `<top (required)>'
	from <internal:/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /home/runner/work/google-cloud-ruby/google-cloud-ruby/google-analytics-admin/test/google/analytics/admin/client_test.rb:19:in `<top (required)>'
	from <internal:/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/rake_test_loader.rb:21:in `block in <main>'
	from /opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/rake_test_loader.rb:6:in `select'
	from /opt/hostedtoolcache/Ruby/3.2.2/x64/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/rake_test_loader.rb:6:in `<main>'
rake aborted!
```

Our test helper basically does this:

```
require "minitest/autorun"
require "minitest/focus"
require "minitest/rg"
```

I determined that setting the `MT_COMPAT=true` environment variable works around this, suggesting this is the change that affected things: https://github.com/minitest/minitest/commit/a2c6c18570f6f0a1bf6af70fe3b6d9599a13fdd6. I'm assuming this means that, unless in compat mode, minitest no longer actually loads plugins until exit time when autorun kicks in, and as a result the code in `minitest/rg` can no longer assume its plugin has already loaded. I guess we could work around this by requiring `minitest/unit` in our helper, but that seems like it isn't the preferred usage, so perhaps RG needs to do something differently?

**Comment by grosser** [comment]
FYI upgrade maxitest to support 5.19 (which vendors minitest/rg) and it just worked 🤷 
possibly because it uses
```
require "minitest"
require "minitest/autorun"
```
but did not dig deeper

**Comment by lustremedia** [comment]
I can confirm this issue. I just switched to maxitest however and it did not work with maxitest as well.

MT_COMPAT=true  solved the issue for me!

**Comment by zenspider** [comment]
`MT_COMPAT=true` or add `require "minitest/unit"` somewhere

**Comment by blowmage** [comment]
@zenspider I haven't dug into what is causing this yet, but would changing the require on [rg_plugin.rb:1](https://github.com/minitest/minitest-rg/blob/v5.2.0/lib/minitest/rg_plugin.rb#L1) from `require "minitest/test"` to `require "minitest"` resolve this?

**Comment by zenspider** [comment]
See #18

**Comment by tsvallender** [comment]
Also seeing this same error (trying to add minitest-rg for the first time), but have been unable to resolve with any of the above suggestions.

**Comment by lustremedia** [comment]
> Also seeing this same error (trying to add minitest-rg for the first time), but have been unable to resolve with any of the above suggestions.

Use maxitest, it has a fix for the issue already -> https://github.com/grosser/maxitest

**Comment by blowmage** [comment]
This fix is released in [minitest-rg 5.3.0](https://rubygems.org/gems/minitest-rg/versions/5.3.0). Thank you all for your patience.
